modules/php/Models/Player.php

```diff
@@ -456,7 +456,9 @@ public function getRange()
   public function hasUnlimitedBangs()
   {
     $weapon = $this->getWeapon();
-    return !is_null($weapon) && $weapon->getType() == CARD_VOLCANIC && !EventCards::getActive()->isBangStrictlyForbidden();
+    $activeEvent = EventCards::getActive();
+    $bangStrictlyForbidden = $activeEvent ? $activeEvent->isBangStrictlyForbidden() : false;
```

Comment:
I think isBangStrictlyForbidden() should be checked in Bang::getPlayOptions() once for all.

Reply:
Makes perfect sense, thanks for checking that!